### PR TITLE
HOC Navbar

### PR DESCRIPTION
**Pull request by shubhamsigdar1**
### Issue

Open #29

### What is the change?
Step1: yarn install which install all dependencies for a project

Step2: files created
1. pages/components/Navbar
     /HOCNavbar.tsx (Higher order component in typescript)
    /navbar.module.scss (style of navbar in sass format)
   /Navbar.tsx (Navbar in next.js)
2. public/icon
    /github-white.png
    /github.png
    /ham.png
3. public/images
   /Avatar.png
   /Real-Dev-squad@1x.png
          
Step3. install the sass package
            npm i sass

### Is Development Tested?

-  No

### Before / After Change Screenshots
hocnavbar

image

image

image

image

### CODE REVIEW

**Comment by shubhamsigdar1** [comment]
@vinayak-trivedi  @vvaibhavdesai

File: package.json [hunk 1]
```diff
@@ -20,6 +20,7 @@
     "react": "18.2.0",
     "react-dom": "18.2.0",
     "react-redux": "^8.0.5",
+    "sass": "^1.58.0",
```
Comment (vinayak-trivedi) [inline package.json:23]: why have we added this?

File: .yarnrc.yml [hunk 2]
```diff
@@ -0,0 +1 @@
+nodeLinker: node-modules
```
Comment (vinayak-trivedi) [inline .yarnrc.yml:1]: why is this here?

**Review by vinayak-trivedi — CHANGES_REQUESTED** [review]
a lot of files are there which shouldn't be there please check ones

File: package.json (hunk 1 above)
Comment (shubhamsigdar1) [reply package.json:23]: /navbar.module.scss (style of navbar in sass format)

File: .yarnrc.yml (hunk 2 above)
Comment (shubhamsigdar1) [reply .yarnrc.yml:1]: I don't know about this

**Comment by shubhamsigdar1** [comment]
> a lot of files are there which shouldn't be there please check ones

These are the files after last commit
1.pages/components/Navbar
      /Layout.tsx (Higher order component in typescript)
      /navbar.module.scss (style of navbar in sass format)
      /Navbar.tsx (Navbar in next.js)
2. public/icon
       /github-white.png
      /github.png
      /ham.png
3. public/images
  /Avatar.png
 /[Real-Dev-squad@1x.png](mailto:Real-Dev-squad@1x.png)

File: package.json (hunk 1 above)
Comment (vinayak-trivedi) [reply package.json:23]: we don't need to do that

File: pages/components/Navbar/Layout.tsx [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import Navbar from './Navbar'
```
Comment (vvaibhavdesai) [inline pages/components/Navbar/Layout.tsx:1]: can you explain what is this `Layout` component for ?

File: pages/components/Navbar/Navbar.tsx [hunk 4]
```diff
@@ -0,0 +1,82 @@
+import React, { useState } from 'react'
+import Image from 'next/image';
+import Link from 'next/link';
+import styles from './navbar.module.scss';
+
+const Navbar = () => {
+  const GITHUB_LOGO = "/icons/github.png"
```
Comment (vvaibhavdesai) [inline pages/components/Navbar/Navbar.tsx:7]: can we use`import` statements ?

File: pages/components/Navbar/Navbar.tsx [hunk 5]
```diff
@@ -0,0 +1,82 @@
+import React, { useState } from 'react'
+import Image from 'next/image';
+import Link from 'next/link';
+import styles from './navbar.module.scss';
+
+const Navbar = () => {
+  const GITHUB_LOGO = "/icons/github.png"
+  const DEFAULT_AVATAR = '/images/Avatar.png';
+  const [toggle, setToggle] = useState(false);
```
Comment (vvaibhavdesai) [inline pages/components/Navbar/Navbar.tsx:9]: please add type for your state variable `useState<boolean>(false)`

File: pages/components/Navbar/Navbar.tsx [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+import Image from 'next/image';
+import Link from 'next/link';
+import styles from './navbar.module.scss';
+
+const Navbar = () => {
+  const GITHUB_LOGO = "/icons/github.png"
+  const DEFAULT_AVATAR = '/images/Avatar.png';
+  const [toggle, setToggle] = useState(false);
+  const sidebarToggle = () => {
+    setToggle(!toggle);
```
Comment (vvaibhavdesai) [inline pages/components/Navbar/Navbar.tsx:11]: please either `setState ` your `state` using `prev=>!prev` or directly by passing `true` or `false`

File: pages/components/Navbar/Navbar.tsx [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+                />
+              </button>
+            </Link>
+          </div>
+          <ul
+            className={`${styles.navBarMenu} ${toggle ? `${styles.active}` : ''
+              }`}
+          >
+            <li className={styles.navBarLogoLi} >
+              <Link href="https://github.com/login/oauth/authorize?client_id=23c78f66ab7964e5ef97">
```
Comment (vvaibhavdesai) [inline pages/components/Navbar/Navbar.tsx:44]: add this link to a constant also why are we shipping this `client ID` ?

File: pages/components/Navbar/Navbar.tsx [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+                <Image
+                  src="/images/Real-Dev-Squad@1x.png"
```
Comment (vvaibhavdesai) [inline pages/components/Navbar/Navbar.tsx:46]: import all the src of the Image in proper variables and use them

**Review by vvaibhavdesai — APPROVED** [review]
also please remove your package changes and don't commit your yarn locks. Change images from png to SVG

**Review by vvaibhavdesai — COMMENTED** [review]
dropped few comments have a look please

File: pages/components/Navbar/Layout.tsx (hunk 3 above)
Comment (shubhamsigdar1) [reply pages/components/Navbar/Layout.tsx:1]: Higher order component

Comment (vvaibhavdesai) [reply pages/components/Navbar/Layout.tsx:1]: what is it for ?

File: pages/components/Navbar/Navbar.tsx [hunk 9]
```diff
@@ -0,0 +1,81 @@
+import { useState } from 'react'
+import Image from 'next/image';
+import Link from 'next/link';
+
+const Navbar = () => {
+  const GITHUB_LOGO = "/icons/github-white.svg";
+  const REALDEVSQUAD_IMAGE="/images/Real-Dev-Squad@1x.svg";
+  let prev=false;
```
Comment (vvaibhavdesai) [inline pages/components/Navbar/Navbar.tsx:8]: what is this prev for ?

File: pages/components/Navbar/Navbar.tsx [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+            <span className="block w-[25px] h-[3px] my-[5px] mx-auto bg-[#fff]"></span>
+            <span className="block w-[25px] h-[3px] my-[5px] mx-auto bg-[#fff]"></span>
+            <span className="block w-[25px] h-[3px] my-[5px] mx-auto bg-[#fff]"></span>
+          </div>
+          <div className="laptop:hidden block m-5 absolute top-0 right-0">
+            <Link
+              className='p-0 m-5.5'
+              href="#"
+            >
+              <button className="rounded-md p-[5px] bg-[#1d1283] text-white cursor-pointer border-2 border-solid border-white flex justify-around" >
```
Comment (vvaibhavdesai) [inline pages/components/Navbar/Navbar.tsx:27]: please add an `onClick` here which has an arrow function and when clicked throws error saying function not declared

File: pages/components/Navbar/Navbar.tsx [hunk 11]
```diff
@@ -0,0 +1,81 @@
+import { useState } from 'react'
+import Image from 'next/image';
+import Link from 'next/link';
+
+const Navbar = () => {
+  const GITHUB_LOGO = "/icons/github-white.svg";
```
Comment (vvaibhavdesai) [inline pages/components/Navbar/Navbar.tsx:6]: please use proper import statements to import the images

File: pages/components/Navbar/Navbar.tsx [hunk 12]
```diff
@@ -0,0 +1,81 @@
...
+              <Link className='laptop:m-0 block laptop:rounded-lg laptop:text-white laptop:text-center laptop:p-0 laptop:no-underline laptop:hover:text-[#49a82e] py-[10px] px-[40px] text-[#1d1283]' href="">
+                <Image
+                  src={REALDEVSQUAD_IMAGE}
+                  alt="home nav logo"
+                  height={55}
+                  width={55}
+                />
+              </Link>
+            </li>
+            <li className="laptop:float-left laptop:hidden block float-none"><Link className='laptop:m-2.5 block laptop:rounded-lg laptop:text-white laptop:text-center laptop:py-5 laptop:px-4 laptop:no-underline hover:text-[#49a82e] py-2.5 px-10 text-[#1d1283]' href="http://realdevsquad.com/">Home</Link></li>
```
Comment (vvaibhavdesai) [inline pages/components/Navbar/Navbar.tsx:53]: can we please use a `map` function instead of writing down `<li/>` so many times ?

File: pages/components/Navbar/Navbar.tsx [hunk 13]
```diff
@@ -0,0 +1,81 @@
...
+            <li className="laptop:float-left float-none"><Link className='laptop:m-2.5 block laptop:rounded-lg laptop:text-white laptop:text-center laptop:py-5 laptop:px-4 laptop:no-underline hover:text-[#49a82e] py-[10px] px-[40px] text-[#1d1283]' href="https://crypto.realdevsquad.com/">Crypto</Link></li>
+            <li className="laptop:float-left float-none"><Link className='laptop:m-2.5 block laptop:rounded-lg laptop:text-white laptop:text-center laptop:py-5 laptop:px-4 laptop:no-underline hover:text-[#49a82e] py-[10px] px-[40px] text-[#1d1283]' href="https://status.realdevsquad.com/">Status</Link></li>
+            <li className="laptop:block laptop:float-right hidden float-none">
+              <Link className='laptop:m-[22px] block laptop:rounded-lg laptop:text-white laptop:text-center laptop:p-0 laptop:no-underline laptop:hover:text-[#49a82e] py-2.5 px-10 text-[#1d1283]'
+                href="#"
+              >
+                <button className="rounded-md p-[5px] bg-[#1d1283] text-white cursor-pointer border-2 border-solid border-white flex justify-around">
```
Comment (vvaibhavdesai) [inline pages/components/Navbar/Navbar.tsx:63]: please add an `onClick `here which has an `arrow function`  and when clicked throws error saying function not declared

**Review by vvaibhavdesai — CHANGES_REQUESTED** [review]
I see unused assets in the code please remove those also the comments are added please resolve those

File: pages/components/Navbar/Layout.tsx (hunk 3 above)
Comment (shubhamsigdar1) [reply pages/components/Navbar/Layout.tsx:1]: High order component of navbar

Comment (shubhamsigdar1) [reply pages/components/Navbar/Layout.tsx:1]: should i change the name?

File: pages/components/Navbar/Navbar.tsx (hunk 4 above)
Comment (shubhamsigdar1) [reply pages/components/Navbar/Navbar.tsx]: No, I removed-> import styles from './navbar.module.scss';

File: pages/components/Navbar/Navbar.tsx (hunk 7 above)
Comment (shubhamsigdar1) [reply pages/components/Navbar/Navbar.tsx]: I removed from code...client id href for now

Comment (shubhamsigdar1) [reply pages/components/Navbar/Navbar.tsx]: from sign-in(mobile) and sign-in with github(desktop)

File: pages/components/Navbar/Navbar.tsx (hunk 11 above)
Comment (shubhamsigdar1) [reply pages/components/Navbar/Navbar.tsx:6]: I didn't understand this vaibhav?
